Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -262,6 +262,55 @@ class BrowserViewController: UIViewController {
                 self.tabManager.addTabAndSelect(request, isPrivate: PrivateBrowsingManager.shared.isPrivateBrowsing)
             }
         }
+        
+        backgroundDataSource.initializeFavorites = { sites in
+            DispatchQueue.main.async {
+                defer { Preferences.NewTabPage.preloadedFavoritiesInitialized.value = true }
+                
+                if Preferences.NewTabPage.preloadedFavoritiesInitialized.value
+                    || Bookmark.hasFavorites { return }
+                
+                guard let sites = sites, sites.count > 0 else {
```

Comment:
```suggestion
                if sites?.isEmpty == true else {
```

Reply:
I need to unwrap `sites`, can eventually change `sites.count > 0` to `!sites.isEmpty` but I dont like the double negative